src/TextHandler.mss

```diff
@@ -111,27 +112,39 @@ function FiguredBassTextHandler (this, textObject) {
 }  //$end
 
 
-function CreateAnchoredText (this, textObj) {
+function CreateAnchoredText (this, textObject) {
     anchoredText = libmei.AnchoredText();
-    AddFormattedText(anchoredText, textObj);
+    AddFormattedText(anchoredText, textObject);
     return anchoredText;
 }  //$end
 
+function CreateDirective (this, textObject) {
+    directive = GenerateControlEvent(textObject, 'Dir');
+    AddFormattedText(directive, textObject);
+    styleIdPrefix = 'text.staff.';
+    text_style = Substring(textObject.StyleId, Length(styleIdPrefix));
+    if (MSplitString(o.StyleId, '.')[-2] = 'user')
+    {
+        text_style = textObject.StyleAsText;
+    }
+    libmei.AddAttribute(directive, 'class', text_style);
```

Comment:
Ah, it should actually be `@type`, right?

Reply:
`@type` would allow it, but there's no formalized specification for what the value of "technique" means. (You could also call it `@type="dinosaur"` and it would have the same level of interoperability.)

Reply:
Having a URI pointing nowhere isn't very nice. I'd prefer `@type`, but if there are spaces in `StyleAsText` that would also be bad. For the time being I switched to `@label` as only attribute allowing plain text.